Sources/SwifterSwift/SwiftStdlib/StringExtensions.swift

```diff
@@ -962,6 +962,22 @@ public extension String {
     }
     #endif
 
+    #if canImport(Foundation)
+    /// SwifterSwift: Verify if string matches the regex pattern and return corresponding parts of string.
+    ///
+    /// - Parameter pattern: Pattern to verify.
+    /// - Returns: Matching pattern parts of string.
+    /// - Throws: Throws any errors thrown by creation of invalid pattern.
+    func matchings(for pattern: String) throws -> [String] {
+        let regex = try NSRegularExpression(pattern: pattern)
+        let results = regex.matches(in: self, range: NSRange(startIndex..., in: self))
```

Comment:
```suggestion
        let results = regex.matches(in: self, range: NSRange(startIndex..<endIndex, in: self))
```
I can't explain why, it's just a hunch, but I feel like explicitly marking the lower and upper bounds will be more performant than the open range. With an open range it feels like there will need to be some search done to find the upper bound, and this can ensure that that doesn't happen.